### PR TITLE
Let File::NULL ("/dev/null", "NUL" etc.) be considered a nil log device

### PR DESCRIPTION
**Pull request by methodmissing**
I noticed on a `Logger` instance with `File::NULL` or `/dev/null` as log device arguments a lot of formatting and other overhead still occurs, with the expected result essentially still being "I want to blackhole these logs":

```
GC: 130 (7.40%)
==================================
     TOTAL    (pct)     SAMPLES    (pct)     FRAME
       348  (19.8%)         348  (19.8%)     Logger::Formatter#format_datetime
       614  (35.0%)         169   (9.6%)     Logger::Formatter#call
       395  (22.5%)         152   (8.7%)     Logger::LogDevice#write
```

The logger was configured as `Logger.new(File::NULL)`.

### CODE REVIEW

File: lib/logger.rb [hunk 1]
```diff
@@ -385,7 +385,7 @@ def initialize(logdev, shift_age = 0, shift_size = 1048576, level: DEBUG,
     self.datetime_format = datetime_format
     self.formatter = formatter
     @logdev = nil
-    if logdev
+    if logdev && log_dev != File::NULL
```
Comment (nobu) [inline lib/logger.rb]: No `_` in the argument name.

Comment (methodmissing) [reply lib/logger.rb]: :man_facepalming: thanks Nobu, fixed.

**Comment by jeremyevans** [comment]
Logger is maintained in a separate repository, this pull request should be submitted there: https://github.com/ruby/logger

**Comment by methodmissing** [comment]
Thanks Jeremy, I’ll PR and explain there

**Comment by methodmissing** [comment]
Rerouted to https://github.com/ruby/logger/pull/47 , closing

**Pull request closed without merging**